tests/foundry/client/canvas/placeables/tokens/ruler.test-d.ts

```diff
@@ -1,4 +1,170 @@
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            !(token instanceof TokenDocument) || !token.hasStatusEffect("prone"),
+          getCostFunction: (
+            _token: TokenDocument.Implementation,
+            _options: foundry.canvas.placeables.Token.MeasureMovementPathOptions,
+          ) => {
+            return (cost: number) => cost * 2;
+          },
+        },
+        crawl: {
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            token instanceof TokenDocument && token.hasStatusEffect("prone"),
+        },
+        fly: {
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            !(token instanceof TokenDocument) || !token.hasStatusEffect("prone"),
+        },
+        jump: {
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            !(token instanceof TokenDocument) || !token.hasStatusEffect("prone"),
+          // default for jump is cost * 2
+          getCostFunction: () => (cost: number) => cost,
+        },
+        swim: {
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            !(token instanceof TokenDocument) || !token.hasStatusEffect("prone"),
+          getCostFunction: (
+            _token: TokenDocument.Implementation,
+            _options: foundry.canvas.placeables.Token.MeasureMovementPathOptions,
+          ) => {
+            return (cost: number) => cost * 2;
+          },
+        },
+        walk: {
+          canSelect: (token: TokenDocument.Implementation | foundry.data.PrototypeToken) =>
+            // eslint-disable-next-line no-restricted-syntax
+            !(token instanceof TokenDocument) || !token.hasStatusEffect("prone"),
+        },
+      },
+      { performDeletions: true },
+    );
+  }
+
+  override _getWaypointLabelContext(
+    waypoint: TokenRuler.Waypoint,
+    state: DrawSteelTokenRuler.State,
+  ): DrawSteelTokenRuler.WaypointContext | void {
+    const context = super._getWaypointLabelContext(waypoint, state) as DrawSteelTokenRuler.WaypointContext | void;
+
+    if (!this.token.inCombat) return context;
+
+    state.segmentWaypoints ??= [];
+    state.segmentWaypoints.push(waypoint);
+
+    if (!context) return;
+
+    const _points = this.token.document.getCompleteMovementPath(state.segmentWaypoints);
+
+    const startedNear = state.endPointEnemies ?? new Set();
+    // Actual code has some Token Document methods to fetch relevant info
+    const endPointEnemies = new Set([]);
+    const passedBy = new Set([]).union(startedNear);
```

Comment:
This is odd code; why not just `const passedBy = new Set(startedNear)` for cloning reasons?

I'm guessing this is just taken from draw-steel.